runtime/vm/classloadersearch.c

```diff
@@ -237,3 +242,52 @@ addZipToLoader(J9JavaVM * vm, const char * filename, J9ClassLoader * classLoader
 
 	return rc;
 }
+
+
+/**
+ * Loads the named module using jdk.internal.module.Modules.loadModule()
+ *
+ * @param [in] vm pointer to JVM
+ * @param [in] moduleName name of module to load
+ * @param [out] modulePtr pointer to return module object.  May be null.
+ *
+ * @return CLS_ERROR_NONEE on success, other value on failure
+ */
+UDATA JNICALL
+loadModule(J9JavaVM *vm, const char *moduleName, jobject *modulePtr)
+{
+	UDATA rc = CLS_ERROR_NONE;
+	J9VMThread *currentThread = currentVMThread(vm);
+	JNIEnv *env = (JNIEnv *) currentThread;
+	jclass jimModules = vm->jimModules;
+
+	if (NULL != modulePtr) {
+		*modulePtr = NULL;
+	}
+
+	/*
+	 * vm->jimModules may have been set already by JVMTI.
+	 * If not, grab a private copy of the class.
+	 */
+	if (NULL == jimModules) {
+		jimModules = (*env)->FindClass(env, "jdk/internal/module/Modules");
+	}
+	if (NULL == jimModules) {
+		rc = CLS_ERROR_NOT_FOUND;
+	} else {
+		jmethodID loadModule =
```

Comment:
Shouldn't this save `jimModules` in `vm->jimModules`?

Reply:
Not clear what you mean here.  If you're going to cache it, you'll need to move the JVMTI code out into a helper in util (definitely don't duplicate the mutexxed init code),

Reply:
Is there a good reason this isn't just in the VM constant pool?  Could then use findKnownClass to get it.

Reply:
That I don't know.  I stole the code from jvmtiModules.c.  Maybe @tajila knows.

Reply:
@tajila Actually, VMCP classes are only loaded eagerly if they have fields/methods declared in the constant pool.  If it's just the class, they are loaded on demand.

Reply:
Discussed with Tobi & Peter, we will go with the CP solution to simplify.